lifting-state-up/single-source-of-truth.md

```diff
@@ -207,9 +221,11 @@ Here, Sofia:
 - Modified how `nameColor` is set: the conditional logic now reads from `props.isPresent`
 - _Commented out_ the `onClick` event handler of the button
 
-Sofia removed the `togglePresence` event handler, so she can't register it with the button anymore. So then what should the button do when it's clicked?
+Sofia removed the `togglePresence` event handler, so she can't register it with the button anymore. So then what should the button do when it's clicked? And what piece of state can the `Student` component update since it no longer contains `isPresent`?
+
+Sofia had previously defined `togglePresence` in `Student` since that was where the piece of state being managed was located. Now that `isPresent` for each student has been moved to `App` (as `isPresentData`), Sofia needs to move the event handler to `App` as well, providing this new handler to each `Student` component as a way to modify the student data. Additionally, she'll need to store the student data itself in a piece of state so that React can be notified when the data changes, and then re-render.
```

Comment:
```suggestion
Sofia had previously defined `togglePresence` in `Student` since that was where the piece of state being managed was located. Now that `isPresent` for each student has been moved to `App` (as `isPresentData`), Sofia needs to move the event handler to `App` as well, providing this new handler to each `Student` component via props as a way to modify the student data. Additionally, she'll need to store the student data itself in a piece of state so that React can be notified when the data changes, and then re-render.
```